src/commands/analyze/utils.ts

```diff
@@ -146,6 +147,49 @@ export async function injectBackendApiUrlsInConfig(configPath: string, frontendP
     await configIOController.write(config);
 }
 
+/**
+ * Injects the SDK into the backend configuration.
+ * @param configPath The path to the config file.
+ */
+export async function injectSDKInConfig(configPath: string) {
+    const configIOController = new YamlConfigurationIOController(configPath);
+
+    // Load config with minimal changes
+    const config = await configIOController.read(/* fillDefaults= */ false);
+
+    const frontend = config.frontend as YamlFrontend;
+    if (!frontend) return;
```

Comment:
I would have expected this to be in an `if` before calling this method.

Reply:
I moved the check - nice catch